### DISCUSSION TITLE
removing images from epub for typst/pdf conversion

### DISCUSSION THREAD

**Discussion by lodwicz** [jgm/pandoc#10299] [Q&A] [answered]
Hello! How can I remove images from epubs in converting to pdf via typst? groff-ms removes images automatically, but typst tries to render them.
```
C:\Downloads>pandoc -o carroll72img.pdf -t typst carroll72img.epub
error: file not found (searched at \\?\C:\Downloads\6012930945704010388_cover.jpg)
    ┌─ \\?\C:\Downloads\toPC185.html:138:11
    │
138 │ #box(image("6012930945704010388_cover.jpg"))
    │            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error producing PDF.
```
Using `--extract-media .` in the CL includes the images, but I only need the placeholders, considering that I may be downloading content from URLs, and [a known vulnerability](https://github.com/jgm/pandoc/security/advisories/GHSA-xj5q-fv23-575g) .

There might be a Lua filter to remove image references, but I can't find it.

carroll72img.epub was downloaded from `https://gutenberg.org/ebooks/12` and renamed to make it intelligible.

**Comment by tarleb** [comment, marked as answer]
This Lua filter should do:

``` lua
funtion Image (img)
    return img.caption
end
```

It replaces the image with the image caption. One could also use `return img.src` to use the image path as the placeholder.
